autoPyTorch/pipeline/components/setup/preprocessor/Preprocessing.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Dict, Optional
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+
+import numpy as np
+
+from autoPyTorch.pipeline.components.setup.base_setup import autoPyTorchSetupComponent
+from autoPyTorch.pipeline.components.setup.preprocessor.utils import get_preprocess_transforms, preprocess
+
+
+class Preprocessing(autoPyTorchSetupComponent):
+
+    def __init__(self, random_state: Optional[np.random.RandomState] = None):
+        self.random_state = random_state
+
+    def fit(self, X: Dict[str, Any], y: Any = None) -> "Preprocessing":
```

Comment:
Shouldn't the preprocessing happen here? I think that fit_transform flow goes like this:

1. fit is called in every component
2. after fit (and here it means we even fitted the network) this preprocessing transform will be called

Sof for the preprocessing to be called I think we need to call  X['train'] = preprocess(dataset=X['train'], transforms=transforms) here

We also need to call preprocessing in the validation data. but I will add a commend on that at the end

Reply:
but then as all the components are added to the fit_dictionary by calling transform on the node, so I think there would not be any preprocessors in the fit_dictionary if this preprocess is done in the fit function.

Reply:
Yeah, please ignore this. I missread:
https://github.com/scikit-learn/scikit-learn/blob/0fb307bf39bbdacd6ed713c00724f8f871d60370/sklearn/pipeline.py#L742

But you are entirely right. I will mark this as resolved